src/js/modules/rpc/server.ts

```diff
@@ -45,6 +54,98 @@ export class ProcessBatchServer extends SupervisorServer {
     }
   }
 
+  enable_coprocessors(input: EnableCoprosRequest): Promise<EnableCoprosReply> {
+    const ids = input.coprocessors.map((script) => script.id);
+    this.logger.info(`request enable wasm script: ${ids}`);
+    const responses = input.coprocessors.map((definition) =>
+      this.validateEnableCoprocInput(definition, this.logger)
+    );
+    return Promise.resolve({ responses });
+  }
+
+  validateEnableCoprocInput(
+    handleDef: EnableCoprocessor,
+    logger: Logger
+  ): EnableCoprocessorRequestData {
+    try {
+      const { id, source_code } = handleDef;
+      const coprocessor = this.loadCoprocFromString(id, source_code);
+      // TODO: Remove Handle type.
+      const handle: Handle = {
+        coprocessor,
+        checksum: "",
+      };
+      const prevHandle = this.repository.findByCoprocessor(handle.coprocessor);
+      if (prevHandle != undefined) {
+        logger.info(`error on load wasm script: ${handleDef.id}`);
+        return errors.createResponseScriptIdAlreadyExists(prevHandle);
+      } else {
+        if (handle.coprocessor.inputTopics.length === 0) {
+          logger.info(`wasm script doesn't have topics: ${handleDef.id}`);
+          return errors.createResponseScriptWithoutTopics(handle);
+        }
+        const validTopics = handle.coprocessor.inputTopics.reduce(
+          (prev, topic) => errors.validateTopic(topic) && prev,
+          true
+        );
+        if (!validTopics) {
+          logger.info(`invalid topic on wasm script: ${handleDef.id}`);
+          return errors.createResponseScriptInvalidTopic(handle);
+        }
+        logger.info(`wasm script loaded on nodejs engine : ${handleDef.id}`);
+        this.repository.add(handle);
+        return errors.createResponseSuccess(handle);
+      }
+    } catch (e) {
+      if (e instanceof SyntaxError) {
+        logger.info(
+          `syntax error on wasm script: ${handleDef.id}, ${e.message}`
+        );
+        return errors.createResponseScriptSyntaxError(handleDef);
+      } else {
+        logger.info(`error on wasm script: ${handleDef.id}, ${e.message}`);
+        return errors.createResponseInternalError(handleDef);
+      }
+    }
+  }
+
+  loadCoprocFromString(id: bigint, script: string): undefined | Coprocessor {
```

Comment:
Is there an optional type in typescript? Maybe instead of `undefined | Coprocessor` it might be better to return `Optional<Coprocessor>`